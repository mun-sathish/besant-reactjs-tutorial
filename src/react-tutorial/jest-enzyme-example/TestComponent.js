import React, { Component } from "react";

export default class TestComponent extends Component {
    render() {
        return (
            <div>
                <h1>Test Component 2</h1>
                <button onClick={this.props.onBtnClicked}>Test Btn</button>
                <p>Prop Text is {this.props.propText}</p>
                <div>
                    <select>
                        <option value="hi1">Option 1</option>
                        <option value="hi2">Option 2</option>
                        <option value="hi3">Option 3</option>
                        <option value="hi4">Option 4</option>
                        <option value="hi5">Option 5</option>
                    </select>
                </div>
            </div>
        );
    }
}
