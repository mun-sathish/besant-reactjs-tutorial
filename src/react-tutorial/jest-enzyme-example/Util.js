export const someFn = (someData) => {
    return "Hi Something" + someData;
};

export const anotherFn = (tag, a, b) => {
    if (tag === "SUM") {
        return a + b;
    } else if (tag === "MIN") {
        return a - b;
    } else if (tag === "MUL") {
        return a * b;
    } else if (tag === "DIV") {
        return a / b;
    } else {
        return -1;
    }
};

export const createObj = (data) => {
    return {
        type: "ADD",
        payload: data,
    };
};
