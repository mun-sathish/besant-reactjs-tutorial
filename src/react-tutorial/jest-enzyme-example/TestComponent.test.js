import React from "react";
import { someFn, anotherFn, createObj } from "./Util";
import renderer from "react-test-renderer";
import TestComponent from "./TestComponent";
import { shallow } from "enzyme";

describe("Test Comp 1", () => {
    it("Test Component H1", () => {
        let wrapper = shallow(<TestComponent />);
        expect(wrapper.find("h1").text()).toBe("Test Component 2");
    });

    it("Test Component Button", () => {
        let wrapper = shallow(<TestComponent />);
        expect(wrapper.find("button").text()).toBe("Test Btn");

        let btnFn = jest.fn();
        let wrapper2 = shallow(<TestComponent onBtnClicked={btnFn} />);
        wrapper2.find("button").first().simulate("click");
        expect(btnFn).toBeCalled();
    });

    it("Test Options", () => {
        let wrapper = shallow(<TestComponent />);
        let count = 0;
        wrapper
            .find("div")
            .first()
            .find("select")
            .first()
            .find("option")
            .forEach((item) => {
                expect(item.text()).toBe("Option " + ++count);
            });

        count = 0;
        wrapper
            .find("div")
            .first()
            .find("select")
            .first()
            .find("option")
            .forEach((item) => {
                expect(item.prop("value")).toBe("hi" + ++count);
            });
    });

    it("Testing some Fn", () => {
        let output = someFn("Sathish");
        expect(output).toBe("Hi SomethingSathish");
    });

    it("Test Snapshot", () => {
        let testCompJS = renderer
            .create(<TestComponent propText={"Coming from Props"} />)
            .toJSON();
        expect(testCompJS).toMatchSnapshot();
    });

    it("Test Obj Snapshot", () => {
        expect(createObj("sampleData")).toMatchSnapshot();
    });

    describe("Testing anotherFn", () => {
        it("Testing for ADD", () => {
            let a = 10;
            let b = 20;
            let output = anotherFn("SUM", a, b);
            expect(output).toBe(a + b);
        });

        it("Testing for SUB", () => {
            let a = 10;
            let b = 20;
            let output = anotherFn("MIN", a, b);
            expect(output).toBe(a - b);
        });

        it("Testing for MUL", () => {
            let a = 10;
            let b = 20;
            let output = anotherFn("MUL", a, b);
            expect(output).toBe(a * b);
        });

        it("Testing for DIV", () => {
            let a = 10;
            let b = 20;
            let output = anotherFn("DIV", a, b);
            expect(output).toBe(a / b);
        });

        it("Testing for Default", () => {
            let a = 10;
            let b = 20;
            let output = anotherFn("Default", a, b);
            expect(output).toBe(-1);
        });
    });
});
