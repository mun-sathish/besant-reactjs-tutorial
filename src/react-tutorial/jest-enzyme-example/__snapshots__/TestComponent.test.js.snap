// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Comp 1 Test Obj Snapshot 1`] = `
Object {
  "payload": "sampleData",
  "type": "ADD",
}
`;

exports[`Test Comp 1 Test Snapshot 1`] = `
<div>
  <h1>
    Test Component 2
  </h1>
  <button>
    Test Btn
  </button>
  <p>
    Prop Text is 
    Coming from Props
  </p>
  <div>
    <select>
      <option
        value="hi1"
      >
        Option 1
      </option>
      <option
        value="hi2"
      >
        Option 2
      </option>
      <option
        value="hi3"
      >
        Option 3
      </option>
      <option
        value="hi4"
      >
        Option 4
      </option>
      <option
        value="hi5"
      >
        Option 5
      </option>
    </select>
  </div>
</div>
`;
